### ISSUE TITLE
OpenCV 4.0.0 does not work with Visual Studio 2019

### ISSUE THREAD

**Issue by rbsheth** [ruslo/hunter#1915]
* I've read [Brief overview section](https://docs.hunter.sh/en/latest/overview.html) and do understand basic concepts. **Yes**
* I've read [F.A.Q. section](https://docs.hunter.sh/en/latest/faq.html) and there is no solution to my problem there. **Yes**
* I've read [Code of Conduct](https://github.com/ruslo/hunter/blob/master/.github/CODE_OF_CONDUCT.md), I promise to be polite and will do my best at being constructive. **Yes**
* I've read [Reporting bugs section](https://docs.hunter.sh/en/latest/contributing.html#reporting-bugs) carefully. **Yes**
* I've checked that all the `hunter_add_package`/`find_package` API used by me in the example is the same as in [documentation](https://docs.hunter.sh/en/latest/packages.html). **Yes**
* I'm using [latest](https://github.com/ruslo/hunter/releases) Hunter `URL`/`SHA1`. **Yes**

* I've created [SSCCE](http://www.sscce.org/) reproducing the issue:
```cmake
# CMakeLists.txt

cmake_minimum_required(VERSION 3.2)

include("cmake/HunterGate.cmake")
HunterGate(
    URL "https://github.com/ruslo/hunter/archive/v0.23.200.tar.gz"
    SHA1 "f6171fde181327a7d3329641bbd543a882222c91"
)

project(foo)

hunter_add_package(OpenCV)
find_package(OpenCV REQUIRED)
```

Here is the log until first error reported by CMake:
```
CMake Warning at <path>/hunter/hunter/_Base/efb5be7/ba1c569/eef6052/Install/OpenCVConfig.cmake:156 (message):
  Found OpenCV Windows Pack but it has no binaries compatible with your
  configuration.

  You should manually point CMake variable OpenCV_DIR to your build of OpenCV
  library.
Call Stack (most recent call first):
  ...

CMake Error at ... (find_package):
  Found package configuration file:

    <path>/hunter/hunter/_Base/efb5be7/ba1c569/eef6052/Install/OpenCVConfig.cmake

  but it set OpenCV_FOUND to FALSE so package "OpenCV" is considered to be
  NOT FOUND.
```

* I'm building on Windows.
* I'm using official CMake release
* CMake version: 3.14.4

* I'm using unmodified toolchain vs-16-2019-win64-cxx17 from [Polly](https://github.com/ruslo/polly)

I'm using the next command line on generate step:
```
cmake -H. -B_builds "-GVisual Studio 16 2019" -A x64
```

Basically, the configure step fails because the version of MSVC is not understood by OpenCV as fixed here upstream: https://github.com/opencv/opencv/commit/8ee1be7d0942bb621b1a98946e32daa6c6ecbb1c#diff-272ceadb8458515b2ae4b5630a6029cc

**Comment by ruslo** [comment]
> Basically, the configure step fails because the version of MSVC is not understood by OpenCV as fixed here upstream: opencv/opencv@8ee1be7#diff-272ceadb8458515b2ae4b5630a6029cc

Since it's an upstream issue, the solution will be to upgrade to OpenCV 4.1 or cherry-pick this fix for current OpenCV 4.0.

**Comment by rbsheth** [comment]
Cool, I see that there is a `hunter-4.1.0` branch in https://github.com/hunter-packages/opencv - is the 4.1.0 release in progress for Hunter?

**Comment by ruslo** [comment]
> I see that there is a hunter-4.1.0 branch

This branch is the same as upstream 4.1.0 tag, feel free to port patches from previous `hunter-4.0.0` branch:
- https://github.com/hunter-packages/opencv/commits/hunter-4.0.0

**Comment by ruslo** [comment]
https://github.com/ruslo/hunter/blob/v0.23.214/README.rst
